alphadia/features.py

```diff
@@ -362,16 +373,76 @@ def center_envelope_1d(x):
 
         for i in range(1, center_index + 1):
             x[a0, center_index - i] = min(left_intensity, x[a0, center_index - i])
+
             left_intensity = (
-                x[a0, center_index - i] + x[a0, center_index - i - 1]
+                x[a0, center_index - i] + x[a0, center_index - i + 1]
             ) * 0.5
 
             x[a0, center_index + i] = min(right_intensity, x[a0, center_index + i])
             right_intensity = (
-                x[a0, center_index + i] + x[a0, center_index + i + 1]
+                x[a0, center_index + i] + x[a0, center_index + i - 1]
             ) * 0.5
 
 
+@nb.njit(inline="always")
+def _even_center_envelope(x: np.ndarray):
+    """
+    Applies an interference correction envelope to a collection of even-length 1D arrays.
+    Numba function which operates in place.
+
+    Parameters
+    ----------
+    x: np.ndarray
+        Array of shape (a, b) where a is the number of arrays and b is the length of each array.
+        It is mandatory that dimension b is even.
+
+    """
+    center_index_right = x.shape[1] // 2
+    center_index_left = center_index_right - 1
+
+    for a0 in range(x.shape[0]):
```

Comment:
why `a0`? just `a` would be less noisy

Reply:
that's a pattern I reuse a lot across numba function for iterating axes. I would keep it as long as I don't remove it completely from the codebase.